src/connection.rs

```diff
@@ -225,9 +216,12 @@ impl Connection {
         self.write_continuation = None;
         self.write_queue.clear();
         self.is_connected = false;
-        let timeout = ServerTimeout::Reconnect(self.token);
+        let timeout = ServerTimeout::Reconnect(token);
         let handle = event_loop.timeout_ms(timeout, duration).unwrap();
-        Ok((duration, timeout, handle))
+
+        info!("{:?}: {:?} reset, will attempt to reconnect in {}ms", self,
```

Comment:
I'm also not in favor of this change.  I remember at the time I wrote this that it was kind of unsavory to return the timeout duration only to be used for logging, but it really does make the log message clearer.  As this is right now, the log message is bogus.